fastddsspy_participants/test/blackbox/model/EndpointDatabaseTest.cpp

```diff
@@ -0,0 +1,418 @@
+    }
+
+    //! Override create_reader() IParticipant method
+    std::shared_ptr<IReader> create_reader(
+            const ITopic& topic)
+    {
+        if (topic.internal_type_discriminator() != spy::participants::INTERNAL_TOPIC_TYPE_ENDPOINT_INFO)
+        {
+            return std::make_shared<ddspipe::participants::BlankReader>();
+        }
+
+        return endpoint_reader;
+    }
+
+    std::shared_ptr<ddspipe::participants::InternalReader> endpoint_reader =
+            std::make_shared<ddspipe::participants::InternalReader>(id_);
+
+};
+
+} // namespace test
+
+std::shared_ptr<DdsPipe> create_pipe(
+        std::shared_ptr<spy::participants::SpyModel>& model,
+        std::shared_ptr<test::SpyDdsParticipantMock>& dds_participant)
+{
+    // Create Payload Pool
+    std::shared_ptr<PayloadPool> payload_pool =
+            std::make_shared<FastPayloadPool>();
+
+    // Create Discovery Database
+    std::shared_ptr<DiscoveryDatabase> discovery_database =
+            std::make_shared<DiscoveryDatabase>();
+
+    // Create and populate Participant Database
+    std::shared_ptr<ParticipantsDatabase> participant_database =
+            std::make_shared<ParticipantsDatabase>();
+
+    // Create participants
+    std::shared_ptr<spy::participants::SpyParticipantConfiguration> spy_configuration =
+            std::make_shared<spy::participants::SpyParticipantConfiguration>();
+    std::shared_ptr<spy::participants::SpyParticipant> spy_participant =
+            std::make_shared<spy::participants::SpyParticipant>(
+        spy_configuration,
+        payload_pool,
+        discovery_database,
+        model);
+
+    participant_database->add_participant(
+        spy_participant->id(),
+        spy_participant
+        );
+
+    participant_database->add_participant(
+        dds_participant->id(),
+        dds_participant
+        );
+
+    // Create allowed topics list
+    std::set<utils::Heritable<types::IFilterTopic>> lists = {};
+    std::shared_ptr<AllowedTopicList> allowed_topics =
+            std::make_shared<AllowedTopicList>(
+        lists,
+        lists);
+
+    // Create Thread Pool
+    unsigned int n_threads = 12;
+    std::shared_ptr<utils::SlotThreadPool> thread_pool =
+            std::make_shared<utils::SlotThreadPool>(n_threads);
+
+    // Create DDS Pipe
+    std::set<utils::Heritable<types::DistributedTopic>> builtin_topics = {};
+    std::shared_ptr<DdsPipe> pipe =
+            std::make_unique<DdsPipe>(
+        allowed_topics,
+        discovery_database,
+        payload_pool,
+        participant_database,
+        thread_pool,
+        builtin_topics,
+        true
+        );
+    return pipe;
+}
+
+void random_endpoint_info(
+        spy::participants::EndpointInfoData& endpoint_data,
+        bool active = true,
+        unsigned int seed = 0,
+        ddspipe::core::types::EndpointKind kind = ddspipe::core::testing::random_endpoint_kind(rand() % 10),
```

Comment:
**In general**:
Try not to use real random values in tests. Use pseudo-randomness. And also, if the method has already a seed, use it instead of calculating a new one.
(I know rand() is not real random, but pseudo-random. But the idea is to have control over what is happening, and not let randomness influences the test.)